mesonbuild/mcompile.py

```diff
@@ -24,9 +26,28 @@
 from .mesonlib import MesonException
 from mesonbuild.environment import detect_ninja
 
-if T.TYPE_CHECKING:
-    import argparse
-    
+def array_arg(value: str) -> T.List[str]:
+    if isinstance(value, str):
+        if value.startswith('['):
+            try:
+                newvalue = ast.literal_eval(value)
+            except ValueError:
+                raise argparse.ArgumentTypeError('malformed option {}'.format(value))
+        elif value == '':
+            newvalue = []
+        else:
+            newvalue = [v.strip() for v in value.split(',')] 
+    elif isinstance(value, list):
+        newvalue = value
+    else:
+        raise argparse.ArgumentTypeError('"{}" should be a string array, but it is not'.format(newvalue))
+
+    for i in newvalue:
+        if not isinstance(i, str):
+            raise argparse.ArgumentTypeError('String array element "{0}" is not a string.'.format(str(newvalue)))
+
+    return newvalue
```

Comment:
I would like to not duplicate that code from UserArrayOption. I think it can be replaced by something like:
```
def array_arg(value: str) -> T.List[str]:
  return coredata.UserArrayOption(None, value, user_input=True).value
```

Reply:
Fixed